utils/debug/PointCloudDebug.js

```diff
@@ -0,0 +1,192 @@
+            if (elt.obj.boxHelper) {
+                elt.obj.boxHelper.material.visible = false;
+            }
+        }
+
+        const stickies = layer.dbgStickyNode.split(',');
+        if (layer.dbgEnableStickyNode) {
+            let keepMe = false;
+            for (const name of stickies) {
+                if (isInHierarchy(elt, name) && // is it either a parent or a child of stickyNode?
+                    (layer.dbgDisplayChildren || elt.name.length <= name.length)) { // is it a child and we don't display them?
+                    keepMe = true;
+                    break;
+                }
+            }
+            if (!keepMe) {
+                return;
+            }
+        }
+        const elms = oldUpdate(context, layer, elt);
+
+        if (layer.dbgEnableStickyNode) {
+            if (stickies.indexOf(elt.name) >= 0) { // is this node the sticky node?
+                layer._currentDbgNode.push(elt);
+                if (elt.obj) {
+                    elt.obj.material.uniforms.useDebugColor.value = true;
+                    elt.obj.material.uniforms.debugColor.value = new itowns.THREE.Color(layer.dbgStickyNodeColor);
+                }
+            } else {
+                for (const name of stickies) {
+                    if (elt.name.length < name.length && name.startsWith(elt.name) && elt.obj) { // is it a parent of sticky node ?
+                        const v = elt.obj.material.visible;
+                        elt.obj.material.visible = layer.dbgDisplayParents;
+
+                        if (v && !elt.obj.material.visible) {
+                            layer.counters.displayedCount -= elt.obj.geometry.drawRange.count;
+                        }
+                    }
+                }
+            }
+        } else if (layer.dbgEnablePointCount) {
+            if (elt.obj && elt.obj.material.visible) {
+                elt.obj.material.uniforms.useDebugColor.value = true;
+
+                const pc = elt.obj.geometry.drawRange.count;
+                if (pc < layer.dbgPointCountThreshold1) {
+                    elt.obj.material.uniforms.debugColor.value.setRGB(1, 0, 0);
+                    if (layer.dbgDisplayType != 'all' && layer.dbgDisplayType != 'red') {
+                        elt.obj.material.visible = false;
+                    }
+                } else if (pc < layer.dbgPointCountThreshold2) {
+                    elt.obj.material.uniforms.debugColor.value.setRGB(0, 1, 0);
+                    if (layer.dbgDisplayType != 'all' && layer.dbgDisplayType != 'green') {
+                        elt.obj.material.visible = false;
+                    }
+                } else {
+                    elt.obj.material.uniforms.debugColor.value.setRGB(0, 0, 1);
+                    if (layer.dbgDisplayType != 'all' && layer.dbgDisplayType != 'blue') {
+                        elt.obj.material.visible = false;
+                    }
+                }
```

Comment:
Let's say you have a scene using 2.5M points. This tool helps you understand the distribution: is there a big node with 2M point, or are there lots of 10k points nodes, etc